app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListView.kt

```diff
@@ -24,37 +24,55 @@ class PromotedStoryListView @JvmOverloads constructor(
 
   @Inject
   lateinit var bindingInterface: ViewBindingShim
+  private var dataList: List<PromotedStoryViewModel> = listOf()
 
   override fun onAttachedToWindow() {
     super.onAttachedToWindow()
 
     (FragmentManager.findFragment<Fragment>(this) as ViewComponentFactory)
       .createViewComponent(this).inject(this)
 
-    adapter = BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
+    /**
+     * The StartSnapHelper is used to snap between items rather than smooth scrolling, so that
+     * the item is completely visible in [HomeFragment] as soon as learner lifts the finger
+     * after scrolling.
+     */
+    val snapHelper = StartSnapHelper()
+    this.onFlingListener = null
+    snapHelper.attachToRecyclerView(this)
+  }
+
+  /**
+   * Sets the list of promoted stories that this view shows to the learner.
+   *
+   * @param newDataList the new list of stories to present
+   */
+  fun setDataList(newDataList: List<PromotedStoryViewModel>) {
+    // Update the adapter and the story list only if the list is new. The parent presenter should
+    // not render promoted stories if the list is empty, but default to showing the last list.
+    if (newDataList != null && !newDataList.isEmpty() && newDataList != dataList) {
+      dataList = newDataList
+      adapter = createAdapter()
+      (adapter as BindableAdapter<PromotedStoryViewModel>).setDataUnchecked(newDataList)
+    }
+  }
+
+  private fun createAdapter(): BindableAdapter<PromotedStoryViewModel> {
+    return BindableAdapter.SingleTypeBuilder.newBuilder<PromotedStoryViewModel>()
       .registerViewBinder(
         inflateView = { parent ->
-          bindingInterface.inflatePromotedStoryCardBinding(
-            inflater = LayoutInflater.from(context),
-            parent = parent,
-            attachToParent = false
+          bindingInterface.providePromotedStoryCardInflatedView(
+            LayoutInflater.from(parent.context),
+            parent,
+            /* attachToParent= */ false
```

Comment:
Should be a named argument, not a comment.

Reply:
Done